'make test-coverage' failed on s390x platform

/kind bug

<!--

Welcome! - We kindly ask you to:

  1. Fill out the issue template below 
  2. Use the Google group if you have a question rather than a bug or feature request.

The group is at: https://groups.google.com/forum/#!forum/odo-users

Thanks for understanding, and for contributing to the project!

-->

## What versions of software are you using?

**Operating System:**
Red Hat Enterprise Linux 8.2 (Ootpa)
**Output of `odo version`:**
odo v1.2.5 (0246cf66c)
Server: https://api.ocp43b.rhocp.boe:6443
Kubernetes: v1.18.3+012b3ec
## How did you run odo exactly?
locally
## Actual behavior
make test-coverage

## Expected behavior
passed

## Any logs, error output, etc?
./scripts/generate-coverage.sh
go test: -race is only supported on linux/amd64, linux/ppc64le, linux/arm64, freebsd/amd64, netbsd/amd64, darwin/amd64 and windows/amd64
make: *** [Makefile:115: test-coverage] Error 2

--- a/scripts/generate-coverage.sh
+++ b/scripts/generate-coverage.sh
@@ -4,11 +4,21 @@
 # go test can't generate code coverage for multiple packages in one command
 
 set -e
+ARCH=$(uname -m)
 echo "" > coverage.txt
-go test -i -race ./cmd/odo
+# The race detector is currently not supported on s390x .
+if [ "${ARCH}" == "s390x" ]; then
+    go test -i ./cmd/odo
+else
+    go test -i -race ./cmd/odo
+fi
+
 for d in $(go list ./... | grep -v vendor | grep -v tests); do
     # For watch related tests, race check causes issue so disabling them here as race is already tested in other tests when used with `-coverprofile=profile.out`
     if [ "$d" = "github.com/openshift/odo/pkg/component" ]; then
+        go test -coverprofile=profile.out -covermode=atomic $d
+    elif [ "${ARCH}" == "s390x" ]; then
+        # The race detector is currently not supported on s390x .
         go test -coverprofile=profile.out -covermode=atomic $d
     else
         go test -race -coverprofile=profile.out -covermode=atomic $d

